types/logger/logger.go

```diff
@@ -42,3 +46,61 @@ func (w funcWriter) Write(p []byte) (int, error) {
 
 // Discard is a Logf that throws away the logs given to it.
 func Discard(string, ...interface{}) {}
+
+type limitData struct {
```

Comment:
document this type and fields. (see https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences and other sections on that page)